File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    let answer = [];
    for(let i = 0; i < queries.length; i++){
        let s = queries[i][0]
        let e = queries[i][1]
        let k = queries[i][2]
        let result = -1;
        for(let j = s; j <= e; j++){
            if(arr[j] > k){
                if(result == -1 || arr[j] < result) result = arr[j];
            }
        }
        answer.push(result);
    }
        
    return answer;
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        for(int i = 0; i < queries.length; i++){
            int s = queries[i][0];
            int e = queries[i][1];
            int k = queries[i][2];
            int result = -1;
            for(int j = s; j <= e; j++){
                if(arr[j] > k){
                    if(result == -1) result = arr[j];
                    else if(arr[j] < result) result = arr[j];
                }
            }
            answer[i] = result;
        }
        return answer;
    }
}